python/GafferUI/AnimationEditor.py

```diff
@@ -313,6 +316,52 @@ def __sourceCurvePlug( self, plug ) :
 		assert( isinstance( result, Gaffer.Animation.CurvePlug ) )
 		return result
 
+	def __animationGadgetContextMenu( self, *unused ) :
+
+		import IECore
+		import functools
+
+		# convert mouse position to event line
+		line = self.__gadgetWidget.getViewportGadget().rasterToGadgetSpace(
+			imath.V2f( GafferUI.Widget.mousePosition( relativeTo = self.__gadgetWidget ) ), gadget = self.__animationGadget )
+
+		# only show context menu when mouse not above axis
+		if self.__animationGadget.onTimeAxis( line ) or self.__animationGadget.onValueAxis( line ) :
+			return False
+
+		# check there are selected keys
+		emptySelectedKeys = not self.__animationGadget.selectedKeys()
+
+		# key interpolation for selected keys
+		interpolation = None if emptySelectedKeys else self.__curveEditor.keyWidget().getInterpolationForSelectedKeys()
+
+		# build context menu
+		menuDefinition = IECore.MenuDefinition()
+
+		for mode in sorted( Gaffer.Animation.Interpolation.values.values() ) :
+			menuDefinition.append(
+				"/Set Interpolation/%s" % ( mode.name ),
```

Comment:
For consistency with the UI language elsewhere, I think this would be "Interpolation" rather than "Set Interpolation". Where the submenu shows some state via checkboxes, the submenu name is usually just the name of the property - see the right click presets menus in the NodeEditor and the various state menus in the Viewer toolbar.